src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticAnalyzerDriver.cs

```diff
@@ -128,14 +128,14 @@ private CompilationWithAnalyzers GetCompilationWithAnalyzers(Compilation compila
             {
                 Interlocked.CompareExchange(
                     ref _lazyCompilationWithAnalyzers,
-                    _owner.GetCompilationWithAnalyzers(_project, compilation, _concurrentAnalysis, _reportSuppressedDiagnostics),
+                    _owner.GetCompilationWithAnalyzers(_project, analyzers, compilation, _concurrentAnalysis, _reportSuppressedDiagnostics),
```

Comment:
This will not work if client of DiagnosticAnalyzerDriver invokes any GetXXXDiagnosticsAsync method with different set of analyzers on the same instance - we will cache _lazyCompilationWithAnalyzers based on first set of analyzers passed it.

I think we should instead pass in the analyzers to the constructor of DiagnosticAnalyzerDriver to make it explicit that each instance of DiagnosticAnalyzerDriver is tied to a set of analyzers and any GetXXXDiagnosticsAsnyc invocation can be called with a specific analyzer from that set.